app/API/V1/Controllers/AuthController.php

```diff
@@ -2,64 +2,174 @@
+
+        /** @var User $user */
+		$user = $this->userRepo->findOneBy(['email'=>$credentials['email']]);
+        if (!Hash::check($credentials['password'], $user->getPassword())) {
+            return response()->json(['error' => trans('auth.invalid_credentials')], 422);
+        }
+
+        if ($user->isActivated() === false) {
+            return response()->json(['error' => trans('auth.email_not_activated')], 403);
+        }
+
+        try {
+			if(($token = $this->auth->attempt($credentials)) === false) {
+				return response()->json(['error' => trans('auth.invalid_credentials')], 401);
 			}
-		} catch(JWTException $e)
-		{
-			return response()->json(['error' => 'could_not_create_token'], 500);
+		} catch(JWTException $e) {
+			return response()->json(['error' => trans('auth.could_not_create_token')], 500);
 		}
-		
+
 		return response()->json(compact('token'));
 	}
-	
+
+    /**
+     * Refresh Tocken
+     *
+     * @return \Illuminate\Http\JsonResponse
+     */
 	public function refresh()
 	{
-		if($this->auth->getToken() === FALSE)
-		{
-			throw new BadRequestHttpException('token_absent');
+		if($this->auth->getToken() === FALSE) {
+			throw new BadRequestHttpException(trans('auth.token_absent'));
 		}
 		
-		try
-		{
+		try {
 			$token = $this->auth->refresh();
-		} catch(TokenInvalidException $e)
-		{
-			throw new UnauthorizedHttpException('token_invalid');
+		} catch(TokenInvalidException $e) {
+			throw new UnauthorizedHttpException(trans('auth.token_invalid'));
 		}
 		
 		return response()->json(compact('token'));
 	}
 
+    /**
+     * @param string $provider
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function getSocialAuth(string $provider)
+    {
+        if(!config("services.$provider"))
+            throw new BadRequestHttpException(trans('auth.no_such_provider'));
+
+        /** @var \Laravel\Socialite\Two\FacebookProvider $redirect */
+        $redirect = $this->socialite->driver($provider)->stateless();
+        $result = [
+            'provider' => $provider,
+            'link' => $redirect->redirect()->getTargetUrl()
+        ];
+
+        return response()->json(compact('result'));
+    }
+
+    /**
+     * @param string $provider
+     * @return \Illuminate\Http\JsonResponse
+     * @throws \Doctrine\DBAL\ConnectionException
+     */
+    public function getSocialAuthCallback(string $provider)
+    {
+        $token = null;
+        if($socializeUser = $this->socialite->driver($provider)->stateless()->user()) {
+            $user = $this->userRepo->findOneBy(['email' => $socializeUser->email]);
+            if (!$user) {
+                /** Generate the user based on the socialite data **/
+                $this->userRepo->registerSocializeUser($provider, $socializeUser);
+                $user = $this->userRepo->findOneBy(['email' => $socializeUser->email]);
+                /** Set user's default role **/
+                $this->roleRepo->setUserPermissions($user);
```

Comment:
It would be good if this method optionally took a list of permission names to apply to the user, but it could default to user.

Also, the name of the method should be: addUserRoles not setUserPermissions. Permissions and roles are different in the system (permissions can be applied to roles. So the name may be confusing.

Reply:
Done. I made it default to ['user']